lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -388,8 +380,58 @@ instance FromBlockfrost BF.ProtocolParams ProtocolParameters where
 instance FromBlockfrost BF.Slot SlotNo where
     fromBlockfrost = fmap SlotNo . (<?#> "SlotNo") . BF.unSlot
 
+instance FromBlockfrost BF.Epoch EpochNo where
+    fromBlockfrost = pure . fromIntegral
+
+
+{-
+┌───────┬───────┬─────────┐
+│ Epoch │ Major │   Era   │
+├───────┼───────┼─────────┤
+│  ...  │   6   │ Alonzo  │
+│  298  │   6   │ Alonzo  │
+├───────┼───────┼─────────┤
+│  297  │   5   │ Alonzo  │
+│  ...  │   5   │ Alonzo  │
+│  290  │   5   │ Alonzo  │
+├───────┼───────┼─────────┤
+│  289  │   4   │  Mary   │
+│  ...  │   4   │  Mary   │
+│  251  │   4   │  Mary   │
+├───────┼───────┼─────────┤
+│  250  │   3   │ Allegra │
+│  ...  │   3   │ Allegra │
+│  236  │   3   │ Allegra │
+├───────┼───────┼─────────┤
+│  235  │   2   │ Shelley │
+│  ...  │   2   │ Shelley │
+│  202  │   2   │ Shelley │
+├───────┼───────┼─────────┤
+│  201  │   1   │  Byron  │
+│  ...  │   1   │  Byron  │
+└───────┴───────┴─────────┘
+-}
+eraByEpoch :: MonadError BlockfrostError m => EpochNo -> m AnyCardanoEra
```

Comment:
```suggestion
eraByEpoch :: EpochNo -> Either BlockfrostError AnyCardanoEra
```

🤔 I like to view Haskell as an onion with different layers: At its core rests the simply typed lambda calculus, then comes parametric polymorphism, then type classes, and further out are rank 2 types (fully polymorphic lambda calculus aka System F), GADTs and type families. I prefer the core layers as a default, and only use outer layers if I can pinpoint the specific advantages I get in return.

(I believe that SPJ has similar views, especially when it comes to the question of dependent types.)

Reply:
I understand your reasoning as: "use the least powerful Haskell that is enough to get a job done".

I was was approaching from a different angle: "maximize the number of contexts where expression is usable [as is]". Same reason why I'd choose `length :: Foldable t => t a -> Int` over `length :: [a] -> Int`. 
In these cases the difference is in the ad-hoc polymorphism. I haven't ever programmed Haskell where it isn't available.